nncf/common/graph/graph.py

```diff
@@ -91,27 +93,27 @@ def layer_attributes(self, value: BaseLayerAttributes) -> None:
 
     @property
     def ignored_algorithms(self) -> List[str]:
-        return self._attributes[NNCFNode.IGNORED_ALGOS_ATTR]
+        return cast(List[str], self._attributes[NNCFNode.IGNORED_ALGOS_ATTR])
 
     def is_in_iteration_scope(self) -> bool:
-        return self._attributes[NNCFNode.IS_IN_ITERATION_SCOPE_NODE_ATTR]
+        return bool(self._attributes[NNCFNode.IS_IN_ITERATION_SCOPE_NODE_ATTR])
 
     def is_integer_input(self) -> bool:
-        return self._attributes[NNCFNode.IS_INTEGER_INPUT_NODE_ATTR]
+        return bool(self._attributes[NNCFNode.IS_INTEGER_INPUT_NODE_ATTR])
 
     def is_shared(self) -> bool:
-        return self._attributes[NNCFNode.IS_SHARED_ATTR]
+        return bool(self._attributes[NNCFNode.IS_SHARED_ATTR])
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return str(self)
 
-    def __str__(self):
+    def __str__(self) -> str:
         return " ".join([str(self.node_id), self.node_name, self.node_type])
 
-    def __hash__(self):
+    def __hash__(self) -> int:
         return hash(str(self))
 
-    def __eq__(self, other):
+    def __eq__(self, other: object) -> bool:
```

Comment:
In this case, it's not feasible to modify the datatype of a `__eq__` parameter due to mismatches with the base class. Since all classes are derived from the object class, which specifies the `__eq__` parameter type as `object`. If we override it, mypy will generate an error.